.github/workflows/ci.yml

```diff
@@ -0,0 +1,102 @@
+name: libsecp256k1-ci
+run-name: libsecp256k1 CI
+on: [pull_request, push]
+
+env:
+  ### compiler options
+  HOST:
+  WRAPPER_CMD:
+  # Specific warnings can be disabled with -Wno-error=foo.
+  # -pedantic-errors is not equivalent to -Werror=pedantic and thus not implied by -Werror according to the GCC manual.
+  WERROR_CFLAGS: '-Werror -pedantic-errors'
+  MAKEFLAGS: '-j4'
+  BUILD: 'check'
+  ### secp256k1 config
+  ECMULTWINDOW: 'auto'
+  ECMULTGENPRECISION: 'auto'
+  ASM: 'no'
+  WIDEMUL: 'auto'
+  WITH_VALGRIND: 'yes'
+  EXTRAFLAGS:
+  ### secp256k1 modules
+  EXPERIMENTAL: 'no'
+  ECDH: 'no'
+  RECOVERY: 'no'
+  SCHNORRSIG: 'no'
+  ELLSWIFT: 'no'
+  ### test options
+  SECP256K1_TEST_ITERS:
+  BENCH: 'yes'
+  SECP256K1_BENCH_ITERS: 2
+  CTIMETESTS: 'yes'
+  # Compile and run the examples.
+  EXAMPLES: 'yes'
+
+jobs:
+  macos-native:
+    name: "x86_64: macOS Ventura"
+    # See: https://github.com/actions/runner-images#available-images.
+    runs-on: macos-13
+
+    env:
+      HOMEBREW_NO_AUTO_UPDATE: 1
+      HOMEBREW_NO_INSTALL_CLEANUP: 1
+      # GitHub Actions give us a fixed number of 4 virtual CPUs. Not that we even have that many jobs at the moment...
+      MAKEFLAGS: '-j5'
+      ASM: 'no'
+      WITH_VALGRIND: 'no'
+      CTIMETESTS: 'no'
+      CC: 'clang'
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - { widemul:  'int64', recovery: 'yes', ecdh: 'yes', schnorrsig: 'yes', ellswift: 'yes' }
```

Comment:
According to https://stackoverflow.com/a/75447432  you can use
```
      matrix:
        include:
          env: ...
```

AFAIU this will avoid all the `env` settings like `BUILD: ${{ matrix.build != '' && matrix.build || env.BUILD }}` below.

Reply:
Reworked.